core/src/test/java/org/apache/calcite/sql/test/SqlOperatorBaseTest.java

```diff
@@ -1595,6 +1598,17 @@ protected static Calendar getCalendarNotTooNear(int timeUnit) {
     tester.checkFails(
         "cast(cast('blah' as varchar(10)) as boolean)", INVALID_CHAR_MESSAGE,
         true);
+
+    tester.checkBoolean("cast(0 as boolean)", false);
+
```

Comment:
Perhaps we could add also the test for the "other int types"?
```
tester.checkBoolean("cast(cast(0 as tinyint) as boolean)", Boolean.FALSE);
tester.checkBoolean("cast(cast(0 as smallint) as boolean)", Boolean.FALSE);
tester.checkBoolean("cast(cast(0 as bigint) as boolean)", Boolean.FALSE);
```

Reply:
got it